libraries/AP_OpenDroneID/AP_OpenDroneID.cpp

```diff
@@ -0,0 +1,759 @@
+     _chan = MAVLINK_COMM_0; //set default to MAVLink channel 0
+     _id_type = MAV_ODID_ID_TYPE_NONE;
+     _ua_type = MAV_ODID_UA_TYPE_NONE;
+     _uav_status = MAV_ODID_STATUS_UNDECLARED;
+    _height_reference = MAV_ODID_HEIGHT_REF_OVER_TAKEOFF;
+
+    _operator_location_type = MAV_ODID_OPERATOR_LOCATION_TYPE_TAKEOFF;
+    _classification_type = MAV_ODID_CLASSIFICATION_TYPE_UNDECLARED;
+    _category_eu = MAV_ODID_CATEGORY_EU_UNDECLARED;
+    _class_eu = MAV_ODID_CLASS_EU_UNDECLARED;
+    _operator_id_type = MAV_ODID_OPERATOR_ID_TYPE_CAA;
+    _description_type = MAV_ODID_DESC_TYPE_TEXT;
+    _operator_latitude = 0.0f;
+    _operator_longitude = 0.0f;
+    _operator_timestamp = 0;
+    _area_count = 1; //default 1
+    _area_radius = 0; //default 0
+    _area_ceiling = -1000; //default -1000
+    _area_floor = -1000; //default -1000
+    _operator_altitude_geo = -1000; //default -1000
+
+    memset(_id_or_mac,0,20);
+    memset(_uas_id,0,20);
+    memset(_operator_id,0,20);
+    memset(_description,0,23);
+
+
+    //code to be replaced: set dummy values in open drone ID messages
+    set_mavlink_channel(MAVLINK_COMM_1); //set to MAVLink channel 1
+    set_mavlink_update_frequency(1); //send dynamic messages a 2 Hz frequency
+    set_id_type(MAV_ODID_ID_TYPE_SERIAL_NUMBER);
+    set_ua_type(MAV_ODID_UA_TYPE_AEROPLANE);
+
+    set_operator_location_type(MAV_ODID_OPERATOR_LOCATION_TYPE_LIVE_GNSS);
+    set_classification_type(MAV_ODID_CLASSIFICATION_TYPE_UNDECLARED);
+    set_category_eu(MAV_ODID_CATEGORY_EU_UNDECLARED);
+    set_class_eu(MAV_ODID_CLASS_EU_UNDECLARED);
+    set_operator_id_type(MAV_ODID_OPERATOR_ID_TYPE_CAA);
+    set_description_type(MAV_ODID_DESC_TYPE_TEXT);
+
+    set_operator_latitude(10.f);
+    set_operator_longitude(30.f);
+    set_operator_timestamp((time_t)(AP_HAL::micros())/1000); //set to current time
+    set_area_count(1); //default 1
+    set_area_radius(0); //default 0
+    set_area_ceiling(-1000); //default -1000
+    set_area_floor(-1000); //default -1000
+    set_operator_altitude_geo(-1000); //default -1000
+
+    uint8_t id_or_mac[20] = "00:00:00:00:00:00";
+    set_id_or_mac(id_or_mac);
+
+    uint8_t uas_id[20] = "SN000000123456789";
+    set_uas_id(uas_id);
+
+    char operator_id[20] = "NLD000000123456789";
+    set_operator_id(operator_id);
+
+    char description[23] = "MAVLink test flight";
+    set_description(description);
+    //end dummy values
+
+    //send static messages
+    send_basic_id_message();
+    send_system_message();
+    send_self_id_message();
+    send_operator_id_message();
+}
+
+void AP_OpenDroneID::update()
+{
+    const uint32_t now = AP_HAL::millis();
+    // send dynamic data to transceiver at frequency defined by _mavlink_update_frequency
+    if (now - _last_send_ms >= (1000/_mavlink_update_frequency)) {
+        _last_send_ms = now;
+        send_dynamic_out();
+    }
+
+    if (_initalized == false) {
+        send_static_out();
+        _initalized = true;
```

Comment:
instead of sending a single time, does it hurt to sent it once every 10 seconds or so?

Reply:
No. except for some extra bytes on the UART link it doesn't hurt.